DEV-INTRO.md

```diff
@@ -1,59 +1,166 @@
 ```
 
-#### "Makefile"
+#### copy web config file
+
+The `docker-compose.dev.yaml` file has a section that will eanble editing the web application. This is by default commented out. If you want to uncoment it you will need to first copy the config.php from the docker/web folder. You can do this using `cp docker/web/config.docker.php web/config.php`.
+
+#### copy R packages
+
+Next copy the R packages from a container to your local machine as the `volumes/lib` folder. This is not really needed, but will speed up the process of the first compilation. Later we will put our newly compiled code here as well. 
+
+You can copy all the data using `docker run -ti --rm -v ${PWD}/volumes/lib:/rlib pecan/base:develop cp -a /usr/local/lib/R/site-library/. /rlib/`. This will copy all compiled packages to your local machine.
+
+This only needs to be done once (or if the PEcAn base image changes drastically, for example a new version of R). You can also always delete all files in the `volumes/lib` folder, and recompile PEcAn from scratch.
+
+### PEcAn Development
+
+To begin development we first have to bring up the full PEcAn stack. This assumes you have done once the steps above. You don't need to stop any running containers, you can use the following command to start all containers: `docker-compose -f docker-compose.yml -f docker-compose.dev.yml up -d`. At this point you have PEcAn running in docker.
+
+The current folder (most likely your clone of the git repository) is mounted in some containers as `/pecan`, and in the case of rstudio also in your home folder as `pecan`. You can see which containers exactly in `docker-compose.dev.yaml`.
+
+You can now modify the code on your local machine, or you can use [rstudio](http://localhost:8000) in the docker stack. Once you made changes to the code you can compile the code either in the terminal of rstudio (`cd pecan && make`) or using `./scripts/compile.sh` from your machine (latter is nothing more than a shell script that runs `docker-compose exec executor sh -c 'cd /pecan && make'`. 
+
+The compiled code is written to `/usr/local/lib/R/site-library` which is mapped to `volumes/lib` on your machine. This same folder is mounted in many other containers, allowing you to share the same PEcAn modules in all containers. Now if you change a module, and compile all other containers will see and use this new version of your module.
 
+To compile the PEcAn code you can use the make command in either the rstudio container, or in the executor container. The script [`compile.sh`](sripts/compile.sh) will run make inside the executor container.
+
+### Workflow Submission
+
+You can submit your workflow either in the executor container or in rstudio container. For example to run the `docker.sipnet.xml` workflow located in the tests folder you can use: 
+
+```
+docker-compose exec executor bash
+# inside the container
+cd /pecan/tests
+R CMD ../web/workflow.R docker.sipnet.xml
```

Comment:
I was able to get this workflow to run! It went until it hit the error below in the writing the config. 
```
2020-05-19 14:42:58 INFO   [write.config.SIPNET] :
   Writing SIPNET configs with input /home/carya/sites/niwot/niwot.clim
Error in gsub("@BINARY@", settings$model$binary, jobsh) :
  invalid 'replacement' argument
Calls: <Anonymous> ... <Anonymous> -> do.call -> write.config.SIPNET -> gsub
```
Is the intention for this sipnet test xml to work for anyone who's gotten the dev environment set up correctly?

Reply:
When I looked at the given model id, there was no binary entry for the model..now I blew everything up I am not able to check again

but either way, the way we set up things -or the pecan code- fails to populate settings$model$binary, normally we shouldn't see this error..I mean I believe it is a docker-related error, but not sure what the exact source is

Reply:
this is not finished yet :( now i need to slowly debug to see what goes wrong.

Reply:
@robkooper gotcha. Can you let us know when it's ready to test?